### PR TITLE
ENH More Actions button is hidden if Campaign is published

### PR DESCRIPTION
**Pull request by sabina-talipova**
## Description
- API endpoints to remove items from campaign return 400 errors when called on a published campaign.
- UI doesn't offer actions to remove items if the campaign has been published.

## Parent issue
- https://github.com/silverstripe/silverstripe-campaign-admin/issues/287

### CODE REVIEW

File: client/src/containers/CampaignAdmin/CampaignAdminList.js [hunk 1]
```diff
@@ -385,15 +385,24 @@ class CampaignAdminList extends Component {
...
+        />
+      )
+      : (
+        <PreviewComponent
+          itemLinks={itemLinks}
+          itemId={itemId}
+          onBack={this.handleCloseItem}
+          className={previewClasses}
+        />
+      );
```
Comment (emteknetnz) [inline client/src/containers/CampaignAdmin/CampaignAdminList.js]:
```suggestion
        const props = {
          itemLinks,
          itemID,
          onBack: this.handleCloseItem,
          className: previewClasses,
        };
        if (State) {
          props.moreActions = this.getMoreActions();
        }
        return <PreviewComponent {...props}/>;
```

File: tests/php/CampaignAdminTest.php [hunk 2]
```diff
@@ -159,4 +161,38 @@ public function testReadCampaign(
         $response = $this->get("/admin/campaigns/set/$changeSetID/show", null, ['Accept' => 'application/json']);
         $this->assertEquals($expectedResponseCode, $response->getStatusCode());
     }
+
+    public function removeCampaignDataProvider()
```
Comment (emteknetnz) [inline tests/php/CampaignAdminTest.php]:
```suggestion
    public function provideRemoveCampaignItem(): array
```

File: tests/php/CampaignAdminTest.php [hunk 3]
```diff
@@ -159,4 +161,38 @@ public function testReadCampaign(
...
+    }
+
+    /**
+     * @dataProvider removeCampaignDataProvider
+     */
+    public function testRemoveCampaignItem(
+        bool $isPublished,
+        bool $isWrongID,
+        int $expectedResponseCode,
+    ) {
```
Comment (emteknetnz) [inline tests/php/CampaignAdminTest.php]:
```suggestion
    /**
     * @dataProvider provideRemoveCampaignItem
     */
    public function testRemoveCampaignItem(
        bool $isPublished,
        bool $isWrongID,
        int $expectedResponseCode,
    ): void {
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Tested locally, works good, some minor code changes

File: tests/php/CampaignAdminTest.php (hunk 2 above)
Comment (sabina-talipova) [reply tests/php/CampaignAdminTest.php]: Updated

File: tests/php/CampaignAdminTest.php (hunk 3 above)
Comment (sabina-talipova) [reply tests/php/CampaignAdminTest.php]: Updated

File: client/src/containers/CampaignAdmin/CampaignAdminList.js (hunk 1 above)
Comment (sabina-talipova) [reply client/src/containers/CampaignAdmin/CampaignAdminList.js]: Updated

**Review by emteknetnz — APPROVED** [review]
Looks good

**Pull request merged**